pkg/api/schema/database_protoconv.go

```diff
@@ -81,7 +88,16 @@ func KVMetadataFromProto(md *KVMetadata) *store.KVMetadata {
 		return nil
 	}
 
-	return store.NewKVMetadata().AsDeleted(md.Deleted)
+	kvmd := store.NewKVMetadata()
+
+	kvmd.AsDeleted(md.Deleted)
+
+	if md.Expiration != nil {
```

Comment:
Is expiration mutually exclusive with deletion? Maybe we could check it here?

Reply:
I think they are not mutually exclusive, expiration is stronger than deletion

Reply:
How should then we interpret a record where there's both deletion and expiration set in metadata?
Should the previous record just be deleted (and expiration being just a delayed delete is covered by such deletion) or does the deletion itself expire and the record reappears after expiration time?

Reply:
Note: deletion and expiration apply to individual entries (not to a key).

deleted entries may still be obtained e.g. history by key or when using embedded store.
expired entries won't be fully readable anymore (actual value is not returned), not even using the embedded store.